direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorageTest.java

```diff
@@ -457,6 +424,74 @@ public boolean onError(Throwable error) {
     handle.close();
   }
 
+  @Test(timeout = 10000)
+  public void testFetchOffsets() throws InterruptedException {
+    InMemStorage storage = new InMemStorage();
+    DataAccessor accessor =
+        storage.createAccessor(
+            direct, createFamilyDescriptor(URI.create("inmem:///inmemstoragetest")));
+    CommitLogReader reader = Optionals.get(accessor.getCommitLogReader(direct.getContext()));
+    AttributeWriterBase writer = Optionals.get(accessor.getWriter(direct.getContext()));
+    long now = System.currentTimeMillis();
+    writer
+        .online()
+        .write(
+            StreamElement.upsert(
+                entity,
+                data,
+                UUID.randomUUID().toString(),
+                "key1",
+                data.getName(),
+                now,
+                new byte[] {1, 2, 3}),
+            (succ, exc) -> {});
+    writer
+        .online()
+        .write(
+            StreamElement.upsert(
+                entity,
+                data,
+                UUID.randomUUID().toString(),
+                "key2",
+                data.getName(),
+                now + 1, // we must ensure the stamp is higher
+                new byte[] {1, 2, 3}),
+            (succ, exc) -> {});
+    Map<Partition, Offset> startingOffsets =
+        reader.fetchOffsets(Position.OLDEST, reader.getPartitions());
+    assertEquals(1, startingOffsets.size());
```

Comment:
Can we have one more test with more partitions?

Reply:
Nice catch, there was actually a bug for the multiple partitions case.